### PR TITLE
Fix selected deployment contract state in Header component

### LINKED ISSUE #201: Fix selected contract deployment state in header component
### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)
As a developer, I want to manage component states properly.

### Description
The selected contract deployment state in the header component has to be reset when a user gets to the home page.

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

Related to https://github.com/MARKETProtocol/dApp/issues/191

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

1. Go to http://dev.dapp.marketprotocol.io/
2. Choose 'Quick deploy' from Deploy Contracts menu
3. Clicking the browser back button to get to the home page.
4. Observe the 'Deploy Contracts' dropdown menu.

### PR DESCRIPTION
**Pull request by arun1595**
##### Checklist
- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Refers/Fixes
Fixes #201

### CODE REVIEW

**Comment by eswarasai** [comment]
@arun1595 -- Can you please rebase with `upstream:develop` and resolve the conflicts? Thanks!

**Comment by arun1595** [comment]
@eswarasai Done!

**Comment by eswarasai** [comment]
@arun1595 -- I think the actual issue here is the `NavItem` `location` props is not getting updated properly on browser back and forward buttons. Can you please take a look at it and update the logic accordingly?

**Comment by arun1595** [comment]
@eswarasai I am sorry, I don't get it.

Can you please elaborate?

**Comment by eswarasai** [comment]
@arun1595 -- Sure. Let's say you visit `Find Contract` and then visit `Test Query`, if you click on browser back button, the state of the Header component is not changed. Ideally, the state of the Header component Menu active is being handled in the `NavItem` based on the `location` props. This needs to be handled for the browser back and forward buttons.

**Comment by arun1595** [comment]
@eswarasai Thanks for the quick response.

> Ideally, the state of the Header component Menu active is being handled in the NavItem based on the location props. 

Aren't we handling the state ourselves? I don't see us using the `NavItem` component. Not sure if I am missing something.

Can you please clarify? Thanks.

**Comment by eswarasai** [comment]
@arun1595 -- You're right. My bad. So earlier we used to use the [`NavItem`](https://github.com/MARKETProtocol/dApp/blob/develop/src/components/NavItem.js) Component. Can you please delete the component as well as the test file associated with `NavItem` and update the PR?

And also, the issue I'm referring to still persists. In simpler terms, the `state` we're maintaining should get updated based on the `window.location.pathname` I presume, if we don't have `location` props.
> Go to Find Contract and then visit Test Query, if you click on browser back button, the state of the Header component Menu item is not changed

**Comment by arun1595** [comment]
@eswarasai My bad, I was only focusing on fixing the state of the selected sub menu items under 'Deploy Contracts'. I've fixed the issue you mentioned, removed the unnecessary `NavItem` component and its test.

Not sure if it is the best solution though. If you know of a better one, let me know.

Thanks!

**Comment by eswarasai** [comment]
@arun1595 -- No worries. So I've updated the PR with the modified logic. Let me know if you approve of it. Thanks :)

**Comment by arun1595** [comment]
@eswarasai LGTM :shipit:
